app/scripts/modules/core/domain/IExecutionStage.ts

```diff
@@ -22,9 +24,13 @@ export interface IExecutionStage extends IOrchestratedItem, IStage {
 }
 
 export interface IExecutionStageSummary extends IExecutionStage {
-  masterStage: IStage;
-  labelTemplate: React.ComponentClass<{ stage: IExecutionStageSummary }>;
+  masterStage: IExecutionStage;
+  stages: IExecutionStage[];
+  labelTemplate: React.ComponentClass<{ stage: IExecutionStageSummary, application?: Application, execution?: IExecution, executionMarker?: boolean }>;
```

Comment:
Generally agreed, but will do this in a separate PR once this is merged for clarity